### PR TITLE
feat(wallet): add support for keystore file (json wallet)

### PR DESCRIPTION
**Pull request by akettal**

### CODE REVIEW

File: packages/mangrove.js/README.md [hunk 1]
```diff
@@ -233,6 +233,15 @@ mgv = await Mangrove.connect('https://mainnet.infura.io/v3/_your_project_id_', {
 mgv = await Mangrove.connect('mainnet' {
   mnemonic: 'clutch captain shoe...', // preferably with environment variable
 });
+
+// Init with a keystore file (json wallet)
+mgv = await Mangrove.connect({
+  provider: 'https://mainnet.infura.io/v3/_your_project_id_',
+  jsonWallet: {
+    path: 'path/to/UTC--created_date_time--address',
+    password: 'wallet_password'
```
Comment (espendk) [inline packages/mangrove.js/README.md:242]: Like for the `privateKey` example above, we should probably note that the password should come from the env.

File: packages/mangrove.js/src/eth.ts [hunk 2]
```diff
@@ -164,8 +174,20 @@ export async function _createSigner(
         "options.signerIndex not applicable to mnemonic, use options.path instead."
       );
     }
+  } else if (options.jsonWallet) {
+    console.log("jsonWallet options");
```
Comment (espendk) [inline packages/mangrove.js/src/eth.ts:178]: Should be deleted, right?

**Review by espendk — CHANGES_REQUESTED** [review]
lgtm, just some minor comments

File: packages/mangrove.js/README.md [hunk 3]
```diff
@@ -233,6 +233,15 @@ mgv = await Mangrove.connect('https://mainnet.infura.io/v3/_your_project_id_', {
 mgv = await Mangrove.connect('mainnet' {
   mnemonic: 'clutch captain shoe...', // preferably with environment variable
 });
+
+// Init with a keystore file (json wallet)
+mgv = await Mangrove.connect({
+  provider: 'https://mainnet.infura.io/v3/_your_project_id_',
+  jsonWallet: {
+    path: 'path/to/UTC--created_date_time--address',
```
[the remark this answers is not in the document]
Comment (akettal) [reply packages/mangrove.js/README.md:241]: Yes, that file name pattern is a practical standard (though it's not documented in an EIP), cf https://geth.ethereum.org/docs/interface/managing-your-accounts. To the point these files are sometimes called "UTC file".

File: packages/mangrove.js/README.md (hunk 1 above)
Comment (akettal) [inline packages/mangrove.js/README.md:242]:
```suggestion
    password: 'wallet_password' // preferably with environment variable
```

File: packages/mangrove.js/src/eth.ts (hunk 2 above)
Comment (akettal) [inline packages/mangrove.js/src/eth.ts:178]:
```suggestion
```

Comment (akettal) [reply packages/mangrove.js/src/eth.ts]: Totally 😓

**Review by espendk — APPROVED** [review]

**Pull request merged**